pkg/handlers/webhookreceiver.go

```diff
@@ -93,49 +100,91 @@ func (h *WebhookReceiverHandler) processAMReceiver(d AMReceiverData, ctx context
 		return &AMReceiverResponse{Error: err, Status: "unable to list managed notifications", Code: http.StatusInternalServerError}
 	}
 
+	// Handle each firing alert
 	for _, alert := range d.Alerts.Firing() {
-		// Ignore alerts which don't have a name
-		alertname, err := alertName(alert)
+		err = h.processAlert(alert, mnl)
 		if err != nil {
-			log.WithError(err).Error("alertname missing for alert")
-			continue
+			log.WithError(err).Error("an alert could not be successfully processed")
 		}
+	}
+	return &AMReceiverResponse{Error: nil, Status: "ok", Code: http.StatusOK}
+}
 
-		// Ignore alerts which don't have the send_managed_notification label
-		if val, ok := alert.Labels[AMLabelManagedNotification]; !ok || val == "false" {
-			log.WithField("alertname", alertname).Debug("ignoring alert with no send_managed_notification label")
-			continue
-		}
+// processAlert handles the pre-check verification and sending of a notification for a particular alert
+// and returns an error if that process completed successfully or false otherwise
+func (h *WebhookReceiverHandler) processAlert(alert template.Alert, mnl *oav1alpha1.ManagedNotificationList) error {
+	// Should this alert be handled?
+	if !isValidAlert(alert) {
+		log.WithField(LogFieldAlert, alert).Info("alert does not meet valid criteria")
+		return fmt.Errorf("alert does not meet valid criteria")
```

Comment:
we log the same message twice?

Reply:
another question, should we return error here? or we just log this as Info and return nil?

Reply:
should this return
````
		return &AMReceiverResponse{Error: err, Status: "alert does not meet valid criteria", Code: http.StatusInternalServerError}
	}
````

Reply:
should this return
````
		return &AMReceiverResponse{Error: err, Status: "alert does not meet valid criteria", Code: http.StatusInternalServerError}
	}
````

Reply:
We could, my main reasoning was whether we ultimately want to do any custom logic back at the `processAlert()` level if the alert is invalid rather than just not needing a notification sent. I originally just had it returning a `bool` to indicate whether a service log was sent or not, then changed it to an error at the last minute. 

Error handling from bad AM input in general is something we still need to address and think about in OA.

Reply:
@T0MASD I disagree, this part of the code should not be caring about the structure of data that is sent back to AlertManager. That should be the handler's job. These functions should be able to run and tested without them needing to know they're interacting with an external client.